metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -88,10 +88,20 @@ final class DefinitionProvider(
     // Find matching symbol occurrence in SemanticDB snapshot
     val occurrence = for {
       queryPosition <- snapshotPosition.toPosition(dirtyPosition.getPosition)
-      occurrence <- snapshot.occurrences.find(
-        _.encloses(queryPosition, includeLastCharacter)
-      )
+      occurrence <- snapshot.occurrences
+        .find(
+          _.encloses(queryPosition, includeLastCharacter)
+        )
+        .orElse(
+          Mtags
+            .allToplevels(source.toInput)
+            .occurrences
+            .find(
+              _.encloses(queryPosition, includeLastCharacter)
```

Comment:
This position doesn’t need to be from edit distance since we parse top levels from the latest text contents of the file.